src/gl/framebuffer.js

```diff
@@ -11,15 +12,19 @@ class Framebuffer {
     colorAttachment: ColorAttachment;
     depthAttachment: DepthAttachment;
 
-    constructor(context: Context, width: number, height: number) {
+    constructor(context: Context, width: number, height: number, hasDepth: boolean) {
         this.context = context;
         this.width = width;
         this.height = height;
         const gl = context.gl;
         const fbo = this.framebuffer = gl.createFramebuffer();
 
         this.colorAttachment = new ColorAttachment(context, fbo);
-        this.depthAttachment = new DepthAttachment(context, fbo);
+        if (hasDepth) {
+            this.depthAttachment = new DepthAttachment(context, fbo);
+        }
+        const status = gl.checkFramebufferStatus(gl.FRAMEBUFFER);
+        assert(status === gl.FRAMEBUFFER_COMPLETE);
```

Comment:
Merging into one line that gets stripped so that we don't get framebuffer status wastefully in production builds.

```suggestion
        assert(gl.checkFramebufferStatus(gl.FRAMEBUFFER) === gl.FRAMEBUFFER_COMPLETE);
```